custom_components/control4/fan.py

```diff
@@ -0,0 +1,179 @@
+"""Platform for Control4 Fan."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from pyControl4.fan import C4Fan
```

Comment:
I realized that pyControl4 doesn't currently have a fan class. Please open a PR there with the fan changes as well. We will need to merge the pyControl4 changes before this integration can be merged.